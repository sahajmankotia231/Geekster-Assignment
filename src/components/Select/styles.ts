import styled from "styled-components";

import { BORDER_GRAY, BLUE, HEAVY_BLACK } from "constants/colors";
import { INPUT_RADIUS } from "constants/styles";
import ArrowDownSvg from "assets/arrow-down.svg";

export const Container = styled.div`
  position: relative;
`;

export const StyledSelect = styled.select`
  appearance: none;
  background: url(${ArrowDownSvg}) no-repeat 90% 8px;
  border: 1px solid ${BORDER_GRAY};
  border-radius: ${INPUT_RADIUS}px;
  color: ${HEAVY_BLACK};
  padding: 12px 44px 12px 16px;

  &:focus {
    border: 1px solid ${BLUE};
    box-shadow: 0 0 1px ${BLUE};
    outline: none;
    transition: all ease 0.2s;
  }
`;
