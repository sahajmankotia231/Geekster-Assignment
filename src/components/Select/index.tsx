import React, { ReactElement, FunctionComponent } from "react";

import { Container, StyledSelect } from "./styles";

interface OptionProps {
  value: string;
}
interface SelectProps {
  onChange: (event) => void;
  value?: string;
  dataTestId?: string;
}

const Select: FunctionComponent<SelectProps> = ({
  children,
  onChange,
  value = "",
  dataTestId = "select",
}): ReactElement => {
  return (
    <Container>
      <StyledSelect data-testid={dataTestId} value={value} onChange={onChange}>
        {children}
      </StyledSelect>
    </Container>
  );
};

export const Option: FunctionComponent<OptionProps> = ({
  children,
  value,
}): ReactElement => <option value={value}>{children}</option>;

export default Select;
