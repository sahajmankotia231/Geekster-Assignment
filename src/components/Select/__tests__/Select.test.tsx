import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import Select, { Option } from "..";

describe("<Select />", () => {
  it("should render properly", () => {
    const { container } = render(
      <Select onChange={() => {}}>
        <Option value="">-</Option>
        <Option value="1">1</Option>
        <Option value="2">2</Option>
      </Select>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should trigger onChange prop", () => {
    const mockedOnChange = jest.fn();

    render(
      <Select onChange={mockedOnChange}>
        <Option value="">-</Option>
        <Option value="1">1</Option>
        <Option value="2">2</Option>
      </Select>
    );

    fireEvent.change(screen.getByTestId("select"), { target: { value: 2 } });

    expect(mockedOnChange).toHaveBeenCalled();
  });
});
