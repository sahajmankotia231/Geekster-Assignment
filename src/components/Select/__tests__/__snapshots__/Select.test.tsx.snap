// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render properly 1`] = `
<div
  class="sc-bdvvaa fKRyFQ"
>
  <select
    class="sc-gsDJrp doAmLO"
    data-testid="select"
  >
    <option
      value=""
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
  </select>
</div>
`;
